rclcpp/include/rclcpp/intra_process_manager.hpp

```diff
@@ -264,7 +283,7 @@ class IntraProcessManager
     // TODO(wjwwood): do something when a message was displaced. log debug?
     (void)did_replace;  // Avoid unused variable warning.
 
-    impl_->store_intra_process_message(intra_process_publisher_id, message_seq);
+    impl_->store_intra_process_message(intra_process_publisher_id, std::move(message_seq));
```

Comment:
That is an error, and doing a std::move here doesn't have any sense. I will correct it.

Reply:
@wjwwood I think it's undefined behavior because passing an rvalue means the caller grants the callee permission to do whatever the callee wants

the following is a dumb example to demonstrate this. the idea is to have two `fn`s, one using lvalue the other rvalue. the rvalue version is justifiable when reusing some memory has some performance benefits -- maybe not in this case but you got the idea

```C++
int fn(const int& a, const int& b) { return a + b; }
int fn(const int& a, int&& b) { b += a; return b; }

void lvalue_call()
{
    int a{3}, b{5};
    int c = fn(a, b);
    std::cout << "[lvalue call] a = " << a << ", b = " << b << ", c = " << c << std::endl;
}

void rvalue_call()
{
    int a{3}, b{5};
    int c = fn(a, std::move(b));
    std::cout << "[rvalue call] a = " << a << ", b = " << b << ", c = " << c << std::endl;
}

```

```
$ g++ -std=c++14 foo.cc && ./a.out 
[lvalue call] a = 3, b = 5, c = 8
[rvalue call] a = 3, b = 8, c = 8
```

the point is, if one casts a value into rvalue, he should not expect anything predictable after that, even when the var is as simple as an `int`

Reply:
Ok, thanks @yyu, looks like we're removing that either way, but I think you're right.